### PR TITLE
Add blueprintjs-core via single package.json w/ npm auto-update 

### PR DESCRIPTION
**Pull request by maruilian11**
Pull request for issue: #9667 
@Piicksarn @cdnjs/intern2 could you please help me review this PR? Thanks!
Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/palantir/blueprint
 * Official website (optional, not the repository): http://blueprintjs.com/
 * NPM package url (optional): https://www.npmjs.com/package/@blueprintjs/core
 * GitHub / Bitbucket popularity (required):
   Watch 100
   Star 4,529
   Fork 130
 * NPM download stats (optional):
   - Downloads in the last day: no data
   - Downloads in the last week: no data
   - Downloads in the last month: no data

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [ ] More than 500 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

**Comment by pvnr0082t** [comment]
LGTM

File: ajax/libs/palantir-blueprint/package.json [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+  "npmName": "@blueprintjs/core",
+  "npmFileMap": [
+    {
+      "basePath": "dist",
+      "files": [
+        "**/*.+(css|js|map)"
+      ]
+    },
+    {
+      "basePath": "resources/icons",
```
Comment (PeterDaveHello) [inline ajax/libs/palantir-blueprint/package.json]: This should not be a base path, you are breaking the structure.

File: ajax/libs/palantir-blueprint/package.json [hunk 2]
```diff
@@ -0,0 +1,121 @@
+{
+  "name": "palantir-blueprint",
+  "version": "1.1.0",
+  "filename": "blueprint.min.css",
+  "description": "A React UI toolkit for the web.",
+  "dependencies": {
```
Comment (PeterDaveHello) [inline ajax/libs/palantir-blueprint/package.json]: Looks like it mixes devDependencies, let's remove it.

**Comment by PeterDaveHello** [comment]
Can we just call it `blueprintjs`? I didn't see a name conflict there.

**Comment by maruilian11** [comment]
wait for the author reply https://github.com/cdnjs/cdnjs/issues/9667#issuecomment-263040401

**Comment by bramus** [comment]
@maruilian11 This PR won't make it work unfortunately. The css files reference `../resources/…`. As they have been moved from their original `./dist` folder to `./` things will break.

**Comment by maruilian11** [comment]
hi @bramus, though `/resources` is not in dist, our auto-update config can catch files from different folders.
you can see our config here:
https://github.com/maruilian11/cdnjs/blob/53d2620f497cfb923507ae2d178710e38a6db1de/ajax/libs/blueprintjs/package.json#L18

**Comment by PeterDaveHello** [comment]
Any updates here? Thanks!

**Comment by maruilian11** [comment]
@PeterDaveHello waiting for the confirmation work finished.
https://github.com/cdnjs/cdnjs/issues/9667#issuecomment-263040442

**Comment by sufuf3** [comment]
@maruilian11 https://github.com/palantir/blueprint/issues/1558#issuecomment-329571787 here is author's answer.
Thank you.

**Comment by giladgray** [comment]
@maruilian11 worth noting that Blueprint has had **27 minor versions** since this PR was opened? 

latest is `1.28.0`. https://github.com/palantir/blueprint/releases

**Comment by maruilian11** [comment]
@giladgray Thanks for the reminding! Our bot will handle that after this PR get merged. :smile: 
@cdnjs/library-reviewer @sufuf3 Please help me review this PR, thanks!

**Comment by maruilian11** [comment]
I will fix this PR later.

**Comment by maruilian11** [comment]
@cdnjs/library-reviewer Please help me review this PR, thanks!

**Comment by maruilian11** [comment]
@extend1994 Yes, I see. Thanks for mention that. How about `blueprintjs-core` since we can't use `blueprintjs/core`.

**Comment by extend1994** [comment]
@maruilian11 Agree with your idea. Would you like to fix this PR?

**Comment by sashberd** [comment]
@extend1994 I helped with lib folder rename

**Comment by sufuf3** [comment]
@sashberd Thank you for your great help.
However, as you know, the lib name is changed, then commit message & PR title should be revised, too.
To save your time, maybe you could check the PR is good enough after you finish it next time.
What do you think?

You are a very good activated-member. I appreciated your time and effort.
Let's put our heads together to improve this project better and better.

Thank you very much.

**Comment by giladgray** [comment]
if i may chine in briefly as the author of `@blueprintjs/core`: the 1.x branch is up to like 1.37 now, and we've shipped several 2.x releases as well. how do upgrades happen on this platform?

also, super fun fact, the repo star count has doubled since this PR was opened a year and a half ago!

**Comment by sashberd** [comment]
@giladgray Thanks for update. Is relevant files both in version>1.28 and version 2.x are still in dist folder and in resources/icons folder? 
Thanks

**Comment by giladgray** [comment]
@sashberd great question, actually things changed quite a bit in 2.0: 
- `dist` now contains only the `*.bundle.js` file
- all the other files are in `lib/`:
  - `lib/css/` contains the css file
  - `lib/{scss,less}/` contain variables files for those preprocessors
  - `lib/cjs/` contains compiled js using CommonJS `require()`  syntax
  - `lib/esm/` contains compiled js using ES modules `import from` syntax
- the icon resources move to a new `@blueprintjs/icons` package

**Comment by sashberd** [comment]
@giladgray Thanks for answer
@sufuf3 @extend1994 Can you help me to figure out how to update for assets icons? Because they are now in another project at all as I understood from @giladgray answer
Thanks

**Comment by sashberd** [comment]
@extend1994 Thus, we can merge current PR, but we have to open issue to add icons for v2 in separate folder?

**Comment by extend1994** [comment]
@sashberd Not pretty sure, let's wait @PeterDaveHello for the decision.

**Comment by sashberd** [comment]
@PeterDaveHello Waiting for your help here

**Comment by sashberd** [comment]
@xtuc  Done. FYR

File: ajax/libs/blueprintjs-core/package.json [hunk 3]
```diff
@@ -0,0 +1,35 @@
+{
+  "name": "blueprintjs-core",
+  "filename": "blueprint.min.css",
```
Comment (MattIPv4) [inline ajax/libs/blueprintjs-core/package.json]:
```suggestion
  "filename": "core.bundle.js",
```

This file does not appear to exist in the most recent version:

```
node_modules/@blueprintjs/core/dist
└── core.bundle.js
```

File: ajax/libs/blueprintjs-core/package.json [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    {
+      "basePath": "dist",
+      "files": [
+        "**/+(blueprint*|core.bundle*)"
+      ]
+    },
+    {
+      "basePath": "",
+      "files": [
+        "resources/icons/*.+(woff|eot|ttf)"
```
Comment (MattIPv4) [inline ajax/libs/blueprintjs-core/package.json]: This does not match anything?

Comment (sashberd) [reply ajax/libs/blueprintjs-core/package.json]: As I see it is exists in version below 3.x, so I think better to fetch this one

**Pull request merged**